tripy/tripy/flat_ir/flat_ir.py

```diff
@@ -48,6 +63,21 @@ def __str__(self):
             layer_strs.append(f"    {str(out)}")
         return "\n".join(layer_strs)
 
+    def optimize(self):
+        """
+        Runs all optimization passes attached to this FlatIR class.
+        """
+        for pass_class in self._passes:
+            pass_instance = pass_class()
+            pass_instance.run(self)
+
+    @classmethod
+    def attach_pass(cls, pass_class):
```

Comment:
Not sure I understand the utility of this method. Why not specify passes directly in the `_passes` member?

Reply:
Ah, I see, we want to be able to add passes out of line. Could we just have a metaclass that does that automatically when inheriting from `FlatIRPass`?